self-core-impl/src/test/java/com/selfxdsd/core/mock/InMemoryContracts.java

```diff
@@ -22,43 +22,392 @@
  */
 package com.selfxdsd.core.mock;
 
-import com.selfxdsd.api.Contract;
-import com.selfxdsd.api.Contracts;
+import com.selfxdsd.api.*;
 import com.selfxdsd.api.storage.Storage;
 
-import java.util.Iterator;
+import java.math.BigDecimal;
+import java.util.*;
+import java.util.stream.StreamSupport;
 
 /**
  * In-Memory contracts for testing purposes.
+ *
  * @author Mihai Andronache (amihaiemil@gmail.com)
  * @version $Id$
  * @since 0.0.1
- * @todo #42:30min Implement this class and write some unit tests
- *  for Project.contracts(). Then go further and implement
- *  Contributor.contracts()
  */
 public final class InMemoryContracts implements Contracts {
-
+    /**
+     * In memory holder of contracts.
+     */
+    private final Set<Contract> contracts = new HashSet<>();
+    /**
+     * In memory holder of projects.
+     */
+    private final Set<ProjectInternal> projects = new HashSet<>();
+    /**
+     * In memory holder of contributors.
+     */
+    private final Set<ContributorInternal> contributors = new HashSet<>();
```

Comment:
@criske Why do we need these 2 sets?
The Project and the Contributor should be taken from the Contract.

Reply:
We need to link a Contract to a Project and Contributor and vice -versa. Those sets are "linked" projects and contributors with their contracts. 
In Contract when we get the Project and then searching in Project#contracts() we must find the same Contract.
Same for Contributor.
See my unit tests.